src/Kestrel.Core/Internal/Http/OutputProducer.cs

```diff
@@ -103,8 +106,8 @@ public void Abort(Exception error)
                 _log.ConnectionDisconnect(_connectionId);
                 _completed = true;
 
-                _pipe.Reader.CancelPendingRead();
-                _pipe.Writer.Complete(error);
+                _outputPipeReader.CancelPendingRead();
```

Comment:
I would love if we could find a way of getting rid of the IPipeReader dependency in this class. Not that this needs to be done as part of this PR, but something to think about.